extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java

```diff
@@ -69,40 +72,16 @@ public boolean canRender(Widget w) {
         }
 
         StringBuilder rowSB = new StringBuilder();
-        for (Mapping mapping : selection.getMappings()) {
-            String rowSnippet = getSnippet("selection_row");
-
-            String command = mapping.getCmd() != null ? mapping.getCmd() : "";
-            String label = mapping.getLabel();
-
-            if (item instanceof NumberItem && ((NumberItem) item).getDimension() != null) {
-                String unit = getUnitForWidget(w);
-                command = StringUtils.replace(command, UnitUtils.UNIT_PLACEHOLDER, unit);
-                label = StringUtils.replace(label, UnitUtils.UNIT_PLACEHOLDER, unit);
-
-                // Special treatment for °C since uom library uses a single character: ℃
-                // This will ensure the current state matches the cmd and the buttonClass is set accordingly.
-                command = StringUtils.replace(command, "°C", "℃");
-            }
-
-            rowSnippet = StringUtils.replace(rowSnippet, "%item%", w.getItem() != null ? w.getItem() : "");
-            rowSnippet = StringUtils.replace(rowSnippet, "%cmd%", StringEscapeUtils.escapeHtml(command));
-            rowSnippet = StringUtils.replace(rowSnippet, "%label%",
-                    label != null ? StringEscapeUtils.escapeHtml(label) : "");
-
-            State compareMappingState = state;
-            if (state instanceof QuantityType) { // convert the item state to the command value for proper
-                                                 // comparison and "checked" attribute calculation
-                compareMappingState = convertStateToLabelUnit((QuantityType<?>) state, command);
+        if (selection.getMappings().size() == 0 && item != null && item.getStateDescription() != null) {
+            for (StateOption option : item.getStateDescription().getOptions()) {
+                String rowSnippet = buildRow(selection, option.getLabel(), option.getValue(), item, state);
+                rowSB.append(rowSnippet);
             }
-
-            if (compareMappingState.toString().equals(command)) {
-                rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "checked=\"true\"");
-                mappedValue = (label != null) ? label : command;
-            } else {
-                rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "");
+        } else {
+            for (Mapping mapping : selection.getMappings()) {
+                String rowSnippet = buildRow(selection, mapping.getLabel(), mapping.getCmd(), item, state);
+                rowSB.append(rowSnippet);
             }
-            rowSB.append(rowSnippet);
         }
         snippet = StringUtils.replace(snippet, "%label_header%", getLabel(w, mappedValue));
```

Comment:
Only a question for my personal understanding: What is the difference between `getLabel(w, mappedValue)` and `getLabel(w)`?
See https://github.com/eclipse/smarthome/pull/5667/files#diff-955d3a6399e421b68151ddb7351de8bbR107